src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java

```diff
@@ -704,6 +709,67 @@ public OptionData addChoice(@Nonnull String name, @Nonnull String value)
         return this;
     }
 
+    /**
+     * Add predefined Choices by Enum-Constants for this OptionData
+     * <br> The user can only provide one of the choices, provided by the enum
+     *
+     * <p> An Enum-Constant can be annotated by {@link OptionChoice} to determine the displayname in the Discord client
+     *
+     * @param enumClass
+     *        The class type of which the Choices shall be added
+
+     * @throws IllegalArgumentException
+     *         If any of the following checks fail
+     *         <ul>
+     *             <li>{@code enumClass} is an enum </li>
+     *             <li>The {@link OptionType} is {@link OptionType#STRING}</li>
+     *             <li>The amount of already set choices is less than {@link #MAX_CHOICES}</li>
+     *             <li>The option is not auto-complete enabled</li>
+     *         </ul>
+     *
+     * @return The OptionData instance, for chaining
+     *
+     * @see net.dv8tion.jda.api.interactions.commands.OptionMapping#getAsEnum(Class)
+     * @see OptionChoice
+     */
+    @Nonnull
+    public OptionData addChoices(@Nonnull Class<?> enumClass)
+    {
+        if (this.choices == null)
```

Comment:
This is unclear code, I would replace it with whether the `OptionType` supports choices, it should be equivalent